conanfile.py

```diff
@@ -317,13 +343,35 @@ def build_configure(self):
                 args.append('--enable-videotoolbox' if self.options.videotoolbox else '--disable-videotoolbox')
                 args.append('--enable-securetransport' if self.options.securetransport else '--disable-securetransport')
 
-            if self.settings.os == "Windows":
+            if self.settings.os == "Windows" and not self._use_mingw:
                 args.append('--enable-libmfx' if self.options.qsv else '--disable-libmfx')
 
+            if tools.cross_building(self.settings):
+                args.append('--enable-cross-compile')
+                args.append('--target-os=%s' % self._target_os)
+                cc = os.environ["CC"]
```

Comment:
cc and cxx should be outside of `if tools.cross_building` (custom compiler may be used even for regular build, e.g. clang on Windows, or custom GCC version on Linux)
it should look like:
```
if `CC` in os.environ:
    args.append(`-cc=%s` % cc)
# same for CXX
```